pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -388,8 +405,65 @@ private void addExtraFiles(File m2Repo) {
+     * with the offliner tool (https://repo1.maven.org/maven2/com/redhat/red/offliner/offliner) are used to download all
+     * the attributes to a local maven repository.
+     */
+
     @Override
     public void close() {
         buildInfoCollector.close();
     }
+
+    // TODO move code to an add-on. Check BXMSPROD-1026
+    private void generateOfflinerManifest() {
+        log.info("Will generate the Offline manifest");
+        List<ArtifactWrapper> artifactsToListRaw = new ArrayList<>();
+        for (PncBuild build : builds.values()) {
+            artifactsToListRaw.addAll(build.getBuiltArtifacts());
+            // TODO: Add filter, basing on the targetRepository.repositoryType, when NCL-6079 is done
+            buildInfoCollector.addDependencies(build, "");
+            artifactsToListRaw.addAll(build.getDependencyArtifacts());
+            log.debug("Dependencies for build {}: {}", build.getId(), build.getDependencyArtifacts().size());
+        }
+        log.debug("Number of collected artifacts for the Offline manifest: {}", artifactsToListRaw.size());
+
+        artifactsToListRaw.removeIf(artifact -> isArtifactExcluded(artifact.getGapv()));
+        log.debug("Number of collected artifacts after exclusion: {}", artifactsToListRaw.size());
+        Map<String, ArtifactWrapper> artifactsToList = new HashMap<>();
+        for (ArtifactWrapper artifact : artifactsToListRaw) {
+            artifactsToList.put(artifact.getGapv(), artifact);
+        }
+        log.info("Number of collected artifacts without duplicates: {}", artifactsToList.size());
+
+        String filename = releasePath + getGenerationData().getOfflineManifestFilename();
+        try (PrintWriter file = new PrintWriter(StandardCharsets.UTF_8.name(), filename)) {
+            for (Map.Entry<String, ArtifactWrapper> artifactEntry : artifactsToList.entrySet()) {
+                ArtifactWrapper artifact = artifactEntry.getValue();
+                // TODO: Remove the check, when NCL-6079 is done
+                if (artifact.getRepositoryType() == RepositoryType.MAVEN) {
+                    GAV gav = artifact.toGAV();
+                    String offlinerString = String
+                            .format("%s,%s/%s", artifact.getSha256(), gav.toVersionPath(), gav.toFileName());
+                    file.println(offlinerString);
+                }
+
+            }
+            List<GAV> extraGavs = generationData.getExternalAdditionalArtifacts()
+                    .stream()
+                    .map(GAV::fromColonSeparatedGAPV)
+                    .collect(Collectors.toList());
+            for (GAV extraGav : extraGavs) {
+                String offlinerString = String.format("%s/%s", extraGav.toVersionPath(), extraGav.toFileName());
+                file.println(offlinerString);
+            }
+
+        } catch (FileNotFoundException | UnsupportedEncodingException e) {
+            log.error("Error generating the Offline manifest", e.getMessage());
```

Comment:
@akoufoudakis This isn't correct. I believe you want `e` not `e.getMessage()`. If you want both, you could so something like:

`log.error("Error generating the offline manifest: {}", e.getMessage(), e);`